paddlenlp/transformers/bloom/modeling.py

```diff
@@ -44,28 +36,64 @@
 from paddlenlp.utils.converter import StateDictNameMapping
 from paddlenlp.utils.log import logger
 
+from .configuration import BloomConfig
+from .processor import (
+    ForcedBOSTokenLogitsProcessor,
+    ForcedEOSTokenLogitsProcessor,
+    HammingDiversityLogitsProcessor,
+    LogitsProcessorList,
+    RepetitionPenaltyLogitsProcessor,
+)
+
+__all__ = [
+    "BloomModel",
+    "BloomForPretraining",
+    "BloomForCausalLM",
+    "BloomForSequenceClassification",
+    "BloomForTokenClassification",
+    "BloomForGeneration",
+]
+
+
 BLOOM_PRETRAINED_MODEL_ARCHIVE_LIST = [
     "bigscience/bloom-560m",
+    "bigscience/bloom-1b1",
+    "bigscience/bloom-3b",
+    "bigscience/bloom-7b",
+    "bigscience/bloom",
+    "bigscience/bloomz-560m",
+    "bigscience/bloomz-1b1",
+    "bigscience/bloomz-3b",
+    "bigscience/bloomz-7b",
+    "bigscience/bloomz",
 ]
 
 
-def parallel_matmul(lm_output, logit_weights, parallel_output=True):
-    hcg = fleet.get_hybrid_communicate_group()
-    model_parallel_group = hcg.get_model_parallel_group()
-    mp_degree = hcg.get_model_parallel_world_size()
-    # rank = hcg.get_model_parallel_rank()
+def parallel_matmul(x: Tensor, y: Tensor, parallel_output=True, mp_degree: int = 1):
```

Comment:
这里为什么要直接传入一个mp_degree，理论上根据环境来适配mp_degree是一个合理的事情

Reply:
我在 comment 当中也写了注释：如果在单卡上执行的话，会报错：

```python
        # if not running under distributed.launch, it will raise AttributeError: 'Fleet' object has no attribute '_hcg'
        hcg = fleet.get_hybrid_communicate_group()
        model_parallel_group = hcg.get_model_parallel_group()
        input_parallel = paddle.distributed.collective._c_identity(x, group=model_parallel_group)
```

所以这里添加了一个`mp_degree`的参数用来判断是不是mp模式。我在这里使用`world_size`来判断了：

```python
def parallel_matmul(x: Tensor, y: Tensor, parallel_output=True):
    world_size = paddle.distributed.get_world_size()
    if world_size > 1:
        # if not running under distributed.launch, it will raise AttributeError: 'Fleet' object has no attribute '_hcg'
        hcg = fleet.get_hybrid_communicate_group()
        model_parallel_group = hcg.get_model_parallel_group()
        input_parallel = paddle.distributed.collective._c_identity(x, group=model_parallel_group)
        logits = paddle.matmul(input_parallel, y, transpose_y=True)
        if parallel_output:
            return logits
        return paddle.distributed.collective._c_concat(logits, group=model_parallel_group)
    else:
        logits = paddle.matmul(x, y, transpose_y=True)
        return logits
```